lib/usdUfe/ufe/UsdClipboardHandler.cpp

```diff
@@ -66,25 +66,61 @@ UsdClipboardHandler::Ptr UsdClipboardHandler::create()
 
 Ufe::UndoableCommand::Ptr UsdClipboardHandler::cutCmd_(const Ufe::Selection& selection)
 {
-    return UsdCutClipboardCommand::create(selection, _clipboard);
+    return UsdUfe::UsdUndoSelectAfterCommand<UsdUfe::UsdCutClipboardCommand>::create(
+        selection, _clipboard);
 }
 
 Ufe::UndoableCommand::Ptr UsdClipboardHandler::copyCmd_(const Ufe::Selection& selection)
 {
     return UsdCopyClipboardCommand::create(selection, _clipboard);
 }
 
+//! \brief Helper class to override the paste command execute
+class UsdPasteClipboardCommandWithSelection
+    : public UsdUndoSelectAfterCommand<UsdPasteClipboardCommand>
+{
+public:
+    using Parent = UsdUndoSelectAfterCommand<UsdPasteClipboardCommand>;
+    using ThisPtr = std::shared_ptr<UsdPasteClipboardCommandWithSelection>;
+
+    // Bring in base class constructors.
+    using Parent::Parent;
+
+    // Override the execute so we can set a selection guard to not erase the
+    // paste as sibling flag when the selection changes as a result of the
+    // paste command selecting the target(s).
+    void execute() override
+    {
+        _clipboard->_inSelectionGuard = true;
+        try {
+            Parent::execute();
+        } catch (...) {
+            _clipboard->_inSelectionGuard = false;
+            throw;
+        }
+        _clipboard->_inSelectionGuard = false;
+    }
```

Comment:
Override the execute method of paste so that we can tell the clipboard to set a selection guard (to ignore the selection change that will come from the paste itself).